lib/SILGen/SILGenBridging.cpp

```diff
@@ -325,6 +325,10 @@ static ManagedValue emitManagedParameter(SILGenFunction &SGF, SILLocation loc,
       return ManagedValue::forBorrowedAddressRValue(value);
     }
 
+  case ParameterConvention::Indirect_In_CXX:
+    // Don't emit a cleanup if the parameter is @in_cxx.
+    return ManagedValue::forOwnedRValue(value, CleanupHandle::invalid());
```

Comment:
@nate-chandler could you take a quick peek at just the SILGen changes to see if the make sense to you?

Reply:
As far as SIL opaque values goes, an easy thing to check would be whether the tests crash out of SILGen with  `-enable-sil-opaque-values`.  Treating such a parameter as owned while suppressing its destruction seems to make sense given the problem being solved.

Reply:
I see crashes when I add `-enable-sil-opaque-values`.

It crashes in `SILValueOwnershipChecker::checkFunctionArgWithoutLifetimeEndingUses` with the error message `Owned function parameter without life ending uses!` when I compile `closure-thunk-executable-macosx.swift`.

This is the SIL that's causing the crash:

```
// thunk for @escaping @callee_guaranteed (@in_guaranteed NonTrivial) -> ()
sil shared [transparent] [serialized] [reabstraction_thunk] [ossa] @$sSo10NonTrivialVIegn_ABIeyBC_TR : $@convention(c) (@inout_aliasable @block_storage @callee_guaranteed (@in_guaranteed NonTrivial) -> (), @in_cxx NonTrivial) -> () {
// %0                                             // user: %2
// %1                                             // user: %5
bb0(%0 : $*@block_storage @callee_guaranteed (@in_guaranteed NonTrivial) -> (), %1 : @owned $NonTrivial):
  %2 = project_block_storage %0 : $*@block_storage @callee_guaranteed (@in_guaranteed NonTrivial) -> () // user: %3
  %3 = load [copy] %2 : $*@callee_guaranteed (@in_guaranteed NonTrivial) -> () // users: %8, %4
  %4 = begin_borrow %3 : $@callee_guaranteed (@in_guaranteed NonTrivial) -> () // users: %6, %5
  %5 = apply %4(%1) : $@callee_guaranteed (@in_guaranteed NonTrivial) -> ()
  end_borrow %4 : $@callee_guaranteed (@in_guaranteed NonTrivial) -> () // id: %6
  %7 = tuple ()                                   // user: %9
  destroy_value %3 : $@callee_guaranteed (@in_guaranteed NonTrivial) -> () // id: %8
  return %7 : $()                                 // id: %9
} // end sil function '$sSo10NonTrivialVIegn_ABIeyBC_TR'
```

This is a thunk function for the block.

It seems that the checker is complaining that the second argument of the thunk (`@in_cxx NonTrivial`) doesn't have lifetime-ending uses.

Reply:
I wonder changing the thunk's parameter to `@in_guaranteed NonTrivial` would fix the crash. Alternatively, maybe `emitObjCThunkArguments` should emit the cleanup instead of suppressing it and instead have `destroy_addr` removed before IRGen.